hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1086,7 +1086,7 @@ public void update(HoodieCommitMetadata commitMetadata, String instantTime) {
           HoodieTableMetadataUtil.convertMetadataToRecords(
               engineContext, dataWriteConfig, commitMetadata, instantTime, dataMetaClient, getTableMetadata(),
               dataWriteConfig.getMetadataConfig(),
-              enabledPartitionTypes, dataWriteConfig.getBloomFilterType(),
+              getMetadataPartitionsToUpdate(), dataWriteConfig.getBloomFilterType(),
```

Comment:
this is the main fix.. good that the added test caught it.

Reply:
Cool. Do we need to change the usage of `enabledPartitionTypes` in other places as well?

Reply:
Also we can store the `getMetadataPartitionsToUpdate` in a variable so that it can be reused later

Reply:
yeah could you please file a followup to refactor the code? Let's not mix it with this PR.